### ISSUE TITLE
Detect the presence of Xcode instead of downloading Xcode Command Line Tools

### ISSUE THREAD

**Issue by iranvir** [Homebrew/brew#11250]
### Provide a detailed description of the proposed feature

When installing Brew, I am forced to install Xcode CLT (command line tools) despite the fact that I already had XCode installed. Given that Xcode already comes with all the necessary tools, I think we should have an option to not download and/or install CLT

### What is the motivation for the feature?

I have the following reasons for this:
1. Using CLT might cause other (non-brew) tools to switch to command line tools as well. I haven't had such issues but I would rather not risk it.
2. Saved disk space. I have a crowded 256GB MacBook and every byte not wasted in redundant packages is helpful.

### How will the feature be relevant to at least 90% of Homebrew users?

I expect not many users have a fully speced out mac, so not wasting storage space might help them all. But honestly, I don't know.

### What alternatives to the feature have been considered?

If detecting the presence of Xcode is not possible. The installation script should just print a message asking if it is installed or not.

**Comment by Bo98** [comment]
There is a couple of formulae which do _require_ the CLT to be installed due to baked in paths, such as Python. Since we have to choose, we prefer requiring the CLT (less than half a GB; fixed location) rather than the full Xcode.app (several GB; variable location), and in the case of older OS (Catalina and earlier) it is even the only option since only the CLT has older SDKs.

**Comment by iranvir** [comment]
Understood! Feel free to close this issue.
Just of out curiousity. Would this cause any conflicts when using tools from Xcode.app in command line or elsewhere? If so, is there a workaround that I could employee?

**Comment by Bo98** [comment]
> Would this cause any conflicts when using tools from Xcode.app in command line

No, the actual binary tools used doesn't matter. They're usually the same between Xcode and the CLT and we can rely on the `/usr/bin` path to allow either to be used.

What does matter is SDKs, e.g. `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk`. For formulae which reference the SDK, then this is the path we use as it's the only one that is in a fixed location.
